src/homework03.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# coding: utf8
+
+"""
+EPAM python q4 homework 03
+Sergey Streltsov 2019-11-09
+"""
+
+
+def list_practice():
+    """List practice
+
+    """
+    src_lst = [i + j for i in 'ab' for j in 'bcd']
+    print(src_lst)
+    new_lst = src_lst[::2]
+    print(new_lst)
+    sec_lst = [str(i) + 'a' for i in range(1, 5)]
+    print(sec_lst)
+    print('{} deleted'.format(sec_lst.pop(1)))
+    thr_lst = sec_lst.copy()
+    thr_lst.insert(1, '2a')
+    print(thr_lst)
+
+
+def tuple_practice():
+    """Tuple practice
+
+    """
+    simple_lst = ['a', 'b', 'c']
+    print(simple_lst)
+    simple_tpl = tuple(simple_lst)
+    print(simple_tpl)
+    lst2 = list(simple_tpl)
+    print(lst2)
+    a, b, c = 'a', 2, 'gamma'
+    print('a={}, b={}, c={}'.format(a, b, c))
+    tpl2 = (('a', 'b', 'c'), )
+    print('{}, length: {}'.format(tpl2, len(tpl2)))
+
+
+def generate_numbers(number=20):
+    """Dictionary practice
+
+    """
+    return {n: n * n for n in range(1, number + 1)}
+
+
+def count_characters(count_me_string):
+    """Dictionary practice
+
+    """
+    return {ch: count_me_string.count(ch) for ch in count_me_string}
```

Comment:
it has O(n**2) complexity. 
There is O(n) solution - please take a look at dicts methods